rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -215,12 +214,27 @@ def __init__(
         # Multicall from MakerDAO: https://github.com/makerdao/multicall/
         self.contract_multicall = contract_multicall
 
-        log.debug(f'Initializing {self.chain_name} inquirer. Nodes to connect {connect_at_start}')
-
         # A cache for erc20 and erc721 contract info to not requery the info
         self.contract_info_erc20_cache: dict[ChecksumEvmAddress, dict[str, Any]] = {}
         self.contract_info_erc721_cache: dict[ChecksumEvmAddress, dict[str, Any]] = {}
-        self.connect_to_multiple_nodes(connect_at_start)
+        self.maybe_connect_to_nodes(when_tracked_accounts=True)
+
+    def maybe_connect_to_nodes(self, when_tracked_accounts: bool) -> None:
+        """Start async connect to the saved nodes for the given evm chain if needed.
+
+        If `when_tracked_accounts` is True then it will connect when we have some
+        tracked accounts in the DB. Otherwise when we have none.
+        """
+        if self.connected_to_any_web3() or self.greenlet_manager.has_task(f'Attempt connection to {self.chain_name} node'):  # noqa: E501
+            return
+
+        with self.database.conn.read_ctx() as cursor:
+            accounts = self.database.get_blockchain_accounts(cursor)
+
+        tracked_accounts_num = len(accounts.get(self.blockchain))
```

Comment:
if I don't have any address in polygon then `get_blockchain_accounts` doesn't return polygon if I udnerstood correctly and then the get here will return `None`? or I missed something

Reply:
Also that function uses `get_key` instead of returning the `SupportedBlockchain` enum value

Reply:
Ahg the get key nvm. Inside is overloaded